src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JRTArchive.java

```diff
@@ -0,0 +1,536 @@
+                                       // Note that REMOVED won't happen since in
+                                       // that case the module listing won't have
+                                       // the resource anyway.
+                                       // Note as well that filter removes files
+                                       // of kind ADDED. Those files are not in
+                                       // the packaged modules, so ought not to
+                                       // get returned from the pipeline.
+                                       return (rd == null ||
+                                               rd.getKind() == ResourceDiff.Kind.MODIFIED);
+                                   })
+                                   .map(s -> {
+                                               String lookupKey = String.format("/%s/%s", module, s);
+                                               return new JRTArchiveFile(JRTArchive.this, s,
+                                                       EntryType.CLASS_OR_RESOURCE,
+                                                       null /* hashOrTarget */,
+                                                       false /* symlink */,
+                                                       resDiff.get(lookupKey));
+                                   })
+                                   .collect(Collectors.toList()));
+            // Finally add all files only present in the resource diff
+            // That is, removed items in the runtime image.
+            files.addAll(resDiff.values().stream()
+                                         .filter(rd -> rd.getKind() == ResourceDiff.Kind.REMOVED)
+                                         .map(s -> {
+                                                 int secondSlash = s.getName().indexOf("/", 1);
+                                                 if (secondSlash == -1) {
+                                                     throw new AssertionError();
+                                                 }
+                                                 String pathWithoutModule = s.getName()
+                                                            .substring(secondSlash + 1,
+                                                                       s.getName().length());
+                                                 return new JRTArchiveFile(JRTArchive.this,
+                                                         pathWithoutModule,
+                                                         EntryType.CLASS_OR_RESOURCE,
+                                                         null  /* hashOrTarget */,
+                                                         false /* symlink */,
+                                                         s);
+                                         })
+                                         .collect(Collectors.toList()));
```

Comment:
```suggestion
                                         .toList());
```

Can replace `.collect(Collectors.toList())` with `Stream::toList` in other occurrances as well.